common/include/scipp/common/except.h

```diff
@@ -24,6 +24,11 @@ template <class T> struct MismatchError : public Error<T> {
   template <class A, class B>
   MismatchError(const A &a, const std::initializer_list<B> b)
       : Error<T>(a, " expected to be equal to one of " + to_string(b)) {}
+
+  template <class A, class B>
+  MismatchError(const A &a, const B &b, const std::string &c)
+      : Error<T>(a, " expected to be equal to " + to_string(b) +
+                        "Mismatched coordinate: " + c) {}
```

Comment:
`MismatchError` is a generic piece of code (in the generic `common/` module/folder) and should therefore not contain anything referencing more specific concepts such as coordinates. This is what I meant on Slack by saying it could be ok to pass the extra message as a whole (`"Mismatched coordinate: " + dim.name()`), or to use the other option of providing `to_string`.